### PR TITLE
New Rocky Paginator

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
- Subclassing Django's Paginator to override some methods, especially removing unnecessary object list counts which reduces api calls. 

### Issue link
https://github.com/minvws/nl-kat-coordination/issues/2598

Closes https://github.com/minvws/nl-kat-coordination/issues/2598

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/rocky/paginator.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        """Validate the given 1-based page number."""
+        try:
+            if isinstance(number, float) and not number.is_integer():
+                raise ValueError
+            number = int(number)
+        except (TypeError, ValueError):
+            raise PageNotAnInteger(_("That page number is not an integer"))
+        if number < 1:
+            raise EmptyPage(_("That page number is less than 1"))
+        return number
```
Comment (ammar92) [inline rocky/rocky/paginator.py:28]: This looks almost identical to the base implementation. Why do we need to override and copy this?

**Review by ammar92 — COMMENTED** [review]
Looks good to me but just one question about validation page numbers

File: rocky/rocky/paginator.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        self,
+        object_list,
+        per_page: int | str,
+        orphans: int = 0,
+        allow_empty_first_page: bool = True,
+    ) -> None:
+        super().__init__(object_list, per_page, orphans, allow_empty_first_page)
+
+        if orphans != 0:
+            raise ValueError("Setting orphans is not supported")
```
Comment (dekkers) [inline rocky/rocky/paginator.py:16]: Using `*args` and `**kwargs` would make this forward compatible with any further changes in Django such as the error_messages argument that is added in Django 5.0.

```suggestion
    def __init__(
        self,
        *args,
        **kwargs,        
    ) -> None:
        super().__init__(*args, **kwargs)

        if self.orphans != 0:
            raise ValueError("Setting orphans is not supported")
```

File: rocky/rocky/paginator.py (hunk 1 above)
Comment (Rieven) [reply rocky/rocky/paginator.py:28]: Yes, we left out the `if number > self.num_pages:
            raise EmptyPage(_("That page contains no results"))` to not do the count again, we don't need that

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Generated multiple pages of findings, tasks and objects. Clicking through the pagination works as expected. Changing the URL parameter also seems to work. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

File: rocky/rocky/views/tasks.py [hunk 3]
```diff
@@ -35,6 +36,7 @@ def get(self, request, *args, **kwargs):
 
 class TaskListView(OrganizationView, ListView):
     paginate_by = 20
+    paginator_class = RockyPaginator
```
Comment (underdarknl) [inline rocky/rocky/views/tasks.py:39]: Is the variable paginator or paginator_class? The docs seems to point at paginator?
https://docs.djangoproject.com/en/5.0/topics/pagination/#using-paginator-in-a-view-function

**Review by dekkers — APPROVED** [review]

**Pull request merged**